### PR TITLE
Update MessageHandlerService to dynamically configure Handler map

### PR DESCRIPTION
**Pull request by RomanTeren**
Add EmailMessageHandler and TemplatedEmailMessageHandler
Add FreeMaker template service to render templates

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/communication/config/CommunicationTenantConfigService.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        super(xmConfigProperties, tenantContextHolder);
+        this.tenantContextHolder = tenantContextHolder;
+    }
+
+    @Override
+    @SneakyThrows
+    public void onRefresh(String updatedKey, String config) {
+        try {
+            super.onRefresh(updatedKey, config);
+            ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
```
Comment (MaximBogun) [inline src/main/java/com/icthh/xm/tmf/ms/communication/config/CommunicationTenantConfigService.java:40]: Should it create every time on refresh?maybe better to create bean.

**Review by MaximBogun — APPROVED** [review]

File: src/main/java/com/icthh/xm/tmf/ms/communication/config/CommunicationTenantConfigService.java (hunk 1 above)
Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/communication/config/CommunicationTenantConfigService.java]: moved objectMapper to a final class field and init it in constructor

File: src/main/java/com/icthh/xm/tmf/ms/communication/messaging/handler/EmailMessageHandler.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        return null;
+    }
+
+    @Override
+    @LogicExtensionPoint(value = "Send", resolver = CustomMessageCreateResolver.class)
+    public CommunicationMessage handle(CommunicationMessageCreate messageCreate) {
+        log.debug("Handling message {}", messageCreate);
+        List<String> emails = messageCreate.getReceiver().stream()
+            .map(Receiver::getEmail)
+            .collect(Collectors.toList());
```
Comment (pkev777) [inline src/main/java/com/icthh/xm/tmf/ms/communication/messaging/handler/EmailMessageHandler.java:43]: JFYI: can be simplified and optimised with 'forEach()'

**Review by pkev777 — APPROVED** [review]

File: src/main/java/com/icthh/xm/tmf/ms/communication/messaging/handler/EmailMessageHandler.java (hunk 2 above)
Comment (RomanTeren) [reply src/main/java/com/icthh/xm/tmf/ms/communication/messaging/handler/EmailMessageHandler.java]: simplified

**Review by icontributor-xm — APPROVED** [review]

**Pull request merged**